src/index.js

```diff
@@ -24,6 +24,11 @@ const isNestedArray = arr => bool(
 
 const isMap = m => bool(Object.prototype.toString.call(m) === '[object Map]')
 
+// check if consumer is yielding our effect to immediatly cause the generator function to throw an error
+const shouldThrowError = obj => bool(obj && Object.keys(obj).includes('@THROW'))
+
+const throwError = message => ({ '@THROW': message })
```

Comment:
I would update `@THROW` to be more like redux-saga's internal convention, just to make this origin of this magic string more clear

RS uses `@@redux-saga/IO`, I'm thinking we should use `@@redux-saga-test-engine/ERROR` for this

Redux saga's objects are more complex, which I don't think is needed here, because we don't do anything with the values yielded from the saga.

```
{
    "@@redux-saga/IO": true,
    "PUT": {
        "channel": null,
        "action": {
            "type": "MY_ACTION_TYPE",
            "payload": {
                "foo": "bar"
            }
        }
    }
}
```